tfhe/src/core_crypto/commons/math/random/activated_random_generator.rs

```diff
@@ -1,19 +1,19 @@
-#[cfg(feature = "generator_x86_64_aesni")]
+#[cfg(all(target_arch = "x86_64", not(feature = "software_prng")))]
 use tfhe_csprng::generators::AesniRandomGenerator;
-#[cfg(feature = "generator_aarch64_aes")]
+#[cfg(all(target_arch = "aarch64", not(feature = "software_prng")))]
 use tfhe_csprng::generators::NeonAesRandomGenerator;
-#[cfg(all(
-    not(feature = "generator_x86_64_aesni"),
-    not(feature = "generator_aarch64_aes")
+#[cfg(any(
+    feature = "software_prng",
+    not(any(target_arch = "x86_64", target_arch = "aarch64"))
 ))]
 use tfhe_csprng::generators::SoftwareRandomGenerator;
 
-#[cfg(feature = "generator_x86_64_aesni")]
+#[cfg(all(target_arch = "x86_64", not(feature = "software_prng")))]
 pub type ActivatedRandomGenerator = AesniRandomGenerator;
-#[cfg(feature = "generator_aarch64_aes")]
+#[cfg(all(target_arch = "aarch64", not(feature = "software_prng")))]
 pub type ActivatedRandomGenerator = NeonAesRandomGenerator;
-#[cfg(all(
-    not(feature = "generator_x86_64_aesni"),
-    not(feature = "generator_aarch64_aes")
+#[cfg(any(
+    feature = "software_prng",
+    not(any(target_arch = "x86_64", target_arch = "aarch64"))
 ))]
 pub type ActivatedRandomGenerator = SoftwareRandomGenerator;
```

Comment:
Should that be part of tfhe-csprng ?

Reply:
I mention that because the same kind of pattern is done in the bench file, so maybe it can be interesting if it's only in one place

Reply:
the activated random is a tfhe-rs construct (it was a concrete-core one before) tfhe-csprng will have several generators available and concrete-core chose the one it liked best to know at compile time which to use which was hidden from the user